tests/components/dyson/test_sensor.py

```diff
@@ -1,355 +1,206 @@
-            _get_with_standby_monitoring(), TEMP_CELSIUS
-        )
-        sensor.hass = self.hass
-        sensor.entity_id = "sensor.dyson_1"
-        assert not sensor.should_poll
-        assert sensor.state == STATE_OFF
-        assert sensor.unit_of_measurement == TEMP_CELSIUS
-        assert sensor.name == "Device_name Temperature"
-        assert sensor.entity_id == "sensor.dyson_1"
-
-    def test_dyson_air_quality_sensor(self):
-        """Test air quality sensor with no value."""
-        sensor = dyson.DysonAirQualitySensor(_get_device_without_state())
-        sensor.hass = self.hass
-        sensor.entity_id = "sensor.dyson_1"
-        assert not sensor.should_poll
-        assert sensor.state is None
-        assert sensor.unit_of_measurement is None
-        assert sensor.name == "Device_name AQI"
-        assert sensor.entity_id == "sensor.dyson_1"
-
-    def test_dyson_air_quality_sensor_with_values(self):
-        """Test air quality sensor with values."""
-        sensor = dyson.DysonAirQualitySensor(_get_with_state())
-        sensor.hass = self.hass
-        sensor.entity_id = "sensor.dyson_1"
-        assert not sensor.should_poll
-        assert sensor.state == 2
-        assert sensor.unit_of_measurement is None
-        assert sensor.name == "Device_name AQI"
-        assert sensor.entity_id == "sensor.dyson_1"
-
-
-@patch("libpurecool.dyson.DysonAccount.login", return_value=True)
-@patch(
-    "libpurecool.dyson.DysonAccount.devices",
-    return_value=[_get_dyson_purecool_device()],
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import entity_registry
+from homeassistant.util.unit_system import IMPERIAL_SYSTEM, METRIC_SYSTEM, UnitSystem
+
+from .common import (
+    BASE_PATH,
+    CONFIG,
+    ENTITY_NAME,
+    NAME,
+    SERIAL,
+    async_update_device,
+    get_basic_device,
 )
-async def test_purecool_component_setup_only_once(devices, login, hass):
-    """Test if entities are created only once."""
-    config = _get_config()
-    await async_setup_component(hass, dyson_parent.DOMAIN, config)
-    await hass.async_block_till_done()
-    discovery.load_platform(hass, "sensor", dyson_parent.DOMAIN, {}, config)
-    await hass.async_block_till_done()
 
-    assert len(hass.data[dyson.DYSON_SENSOR_DEVICES]) == 4
+from tests.common import async_setup_component
+
+ENTITY_ID_PREFIX = f"{PLATFORM_DOMAIN}.{ENTITY_NAME}"
+
+MOCKED_VALUES = {
+    "filter_life": 100,
+    "dust": 5,
+    "humidity": 45,
+    "temperature_kelvin": 295,
+    "temperature": 21.9,
+    "air_quality": 5,
+    "hepa_filter_state": 50,
+    "combi_filter_state": 50,
+    "carbon_filter_state": 10,
+}
+
+MOCKED_UPDATED_VALUES = {
+    "filter_life": 30,
+    "dust": 2,
+    "humidity": 80,
+    "temperature_kelvin": 240,
+    "temperature": -33.1,
+    "air_quality": 3,
+    "hepa_filter_state": 30,
+    "combi_filter_state": 30,
+    "carbon_filter_state": 20,
+}
+
+SENSOR_ATTRIBUTES = {
```

Comment:
It would be great if we could move this to the production code and use the same lookup table in the production code. Could we make the values dicts with the attribute names as keys and the corresponding attribute values as values? Then we should be able to use this map also in the production code.

The benefit is that we wouldn't need to change the test when adding or changing sensors.

Reply:
Done. I use `device_state_attributes` to achieve this. The attribute dictionaries are combined.